PKHeX.Core/Legality/Encounters/EncounterGenerator.cs

```diff
@@ -129,7 +129,7 @@ private static IEnumerable<IEncounterable> GenerateFilteredEncountersPIDIV(PKM p
         private static IEnumerable<IEncounterable> GenerateRawEncounters(PKM pkm)
         {
             int ctr = 0;
-            if (pkm.WasLink)
+            if (pkm.WasLink && pkm.GenNumber == 6)
             {
                 foreach (var z in getValidLinkGifts(pkm))
```

Comment:
In that case, that begs the question why a switch statement is used in getValidLinkGifts.

Oh, well. It works either way.